pkgs/by-name/as/astro-language-server/package.nix

```
@@ -0,0 +1,100 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  stdenvNoCC,
+  jq,
+  moreutils,
+  nodePackages,
+  cacert,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "astro-language-server";
+  version = "2.9.0";
+
+  src = fetchFromGitHub {
+    owner = "withastro";
+    repo = "language-tools";
+    rev = "@astrojs/language-server@${finalAttrs.version}";
+    hash = "sha256-OfPe2bGcYWXZHkFc0iCiU3dPxbSsxX+ZVN6Fd7bMf9o=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/packages/language-server";
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${finalAttrs.pname}-pnpm-deps";
+    inherit (finalAttrs) src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    # https://github.com/NixOS/nixpkgs/blob/de80f1eeac3152c5bbfb1f8891b6414d526bfc54/pkgs/by-name/po/pot/package.nix#L54
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # use --ignore-script and --no-optional to avoid downloading binaries
+      # use --frozen-lockfile to avoid checking git deps
+      pnpm install --frozen-lockfile --no-optional --ignore-script
```

```suggestion
      pnpm install --frozen-lockfile --no-optional --ignore-script --force
```
this is the result of the hash mismatch. without `--force`, pnpm will only install dependencies for the current architecture. we might want to extract from https://github.com/NixOS/nixpkgs/pull/290715 or `vesktop` instead

---

Edited this in and fixed the hash. Thanks for catching this!